cli/storage_profile.c

```diff
@@ -236,7 +236,7 @@ perform_profile_run(
     sigaddset(&blockset, SIGALRM);
     pthread_sigmask(SIG_BLOCK, &blockset, &origset);
 
-    work_specs = malloc(thread_cnt * sizeof(struct storage_profile_work));
+    work_specs = calloc(thread_cnt, sizeof(struct storage_profile_work));
```

Comment:
We don't necessarily want to replace all `malloc(a*b)` with `calloc(a,b)`.  We have a wrapper `malloc_array(a,b)` that gives use the two param API w/o initializing memory.

Reply:
What was clang-tidy's complaint?

Reply:
The complaint was that the `free(work_specs[i].samples)` down below had the potential to free uninitialized memory, so this was the easiest solution.

Reply:
DIdn't you just suggest getting rid of malloc_array() as there was only one caller?

Reply:
I did. :)

Reply:
I thought clang-tidy might have been complaining that calloc should be used.

Reply:
I take it you mean accesing unitialized memory, there shouldn't be any issue with freeing uninitialized memory...

Reply:
Well if I have a pointer that points to garbage, is that not uninitialized memory?

If I say `free((void *)1)`, this would definitely error out.

Reply:
Right, that's freeing a ptr you didn't get from malloc vs "freeing uninitialized memory", e.g.,    p = malloc(100); free(p);

Reply:
Hmm I guess you're right :)